File: python/lexer_test.py
import unittest
from .token import TokenType
class TestLexer(unittest.TestCase):
    def test_next_token(self):
        input = "=+(){},;"
        tests = [
            (TokenType.ASSIGN, "="),
            (TokenType.PLUS, "+"),
            (TokenType.LPAREN, "("),
            (TokenType.RPAREN, ")"),
            (TokenType.LBRACE, "{"),
            (TokenType.RBRACE, "}"),
            (TokenType.COMMA, ","),
            (TokenType.SEMICOLON, ";"),
            (TokenType.EOF, ""),
        ]
        
        lexer = Lexer(input)

        for i, (expected_type, expected_literal) in enumerate(tests):
            token = lexer.next_token()
            self.assertEqual(token.type, expected_type, f"tests[{i}] - tokentype wrong. expected={expected_type}, got={token.type}")
            self.assertEqual(token.literal, expected_literal, f"tests[{i}] - literal wrong. expected={expected_literal}, got={token.literal}")

if __name__ == "__main__":
    unittest.main()

File: ts/tests/lexer.test.ts
// Importing the necessary modules and types
import { Lexer } from '../src/lexer'; // Assume you have a lexer.ts file that exports a Lexer class
import { TokenType, Token, ASSIGN, PLUS, LPAREN, RPAREN, LBRACE, RBRACE, COMMA, SEMICOLON, EOF } from '../src/token'; // Assume you have a token.ts file that exports TokenType enum and Token interface


describe('Lexer', () => {
  test('NextToken', () => {
    const input = '=+(){},;';
    const tests: { expectedType: TokenType; expectedLiteral: string }[] = [
      { expectedType: ASSIGN, expectedLiteral: '=' },
      { expectedType: PLUS, expectedLiteral: '+' },
      { expectedType: LPAREN, expectedLiteral: '(' },
      { expectedType: RPAREN, expectedLiteral: ')' },
      { expectedType: LBRACE, expectedLiteral: '{' },
      { expectedType: RBRACE, expectedLiteral: '}' },
      { expectedType: COMMA, expectedLiteral: ',' },
      { expectedType: SEMICOLON, expectedLiteral: ';' },
      { expectedType: EOF, expectedLiteral: '' },
    ];

    const lexer = new Lexer(input);

    tests.forEach((tt, i) => {
      const tok = lexer.nextToken();
      expect(tok.type).toBe(tt.expectedType);
      expect(tok.literal).toBe(tt.expectedLiteral);
    });
  });
});
